stubs/SQLAlchemy/sqlalchemy/engine/base.pyi

```diff
@@ -1,61 +1,97 @@
         _dispatch: Any | None = ...,
         _has_events: Any | None = ...,
         _allow_revalidate: bool = ...,
     ) -> None: ...
-    def schema_for_object(self, obj): ...
-    def __enter__(self): ...
-    def __exit__(self, type_, value, traceback) -> None: ...
+    def schema_for_object(self, obj) -> str: ...
+    def __enter__(self: Self) -> Self: ...
+    def __exit__(
+        self, type_: type[BaseException] | None, value: BaseException | None, traceback: TracebackType | None
+    ) -> None: ...
     def execution_options(self, **opt): ...
     def get_execution_options(self): ...
     @property
-    def closed(self): ...
+    def closed(self) -> bool: ...
     @property
-    def invalidated(self): ...
+    def invalidated(self) -> bool: ...
     @property
-    def connection(self): ...
+    def connection(self) -> DBAPIConnection: ...
     def get_isolation_level(self): ...
     @property
     def default_isolation_level(self): ...
     @property
     def info(self): ...
     def connect(self, close_with_result: bool = ...): ...  # type: ignore[override]
-    def invalidate(self, exception: Any | None = ...): ...
+    def invalidate(self, exception: Exception | None = ...) -> None: ...
     def detach(self) -> None: ...
-    def begin(self): ...
-    def begin_nested(self): ...
-    def begin_twophase(self, xid: Any | None = ...): ...
+    def begin(self) -> Transaction | None: ...
+    def begin_nested(self) -> Transaction | None: ...
+    def begin_twophase(self, xid: Any | None = ...) -> TwoPhaseTransaction: ...
     def recover_twophase(self): ...
     def rollback_prepared(self, xid, recover: bool = ...) -> None: ...
     def commit_prepared(self, xid, recover: bool = ...) -> None: ...
-    def in_transaction(self): ...
-    def in_nested_transaction(self): ...
-    def get_transaction(self): ...
-    def get_nested_transaction(self): ...
+    def in_transaction(self) -> bool: ...
+    def in_nested_transaction(self) -> bool: ...
+    def get_transaction(self) -> Transaction | None: ...
+    def get_nested_transaction(self) -> Transaction | None: ...
     def close(self) -> None: ...
-    def scalar(self, object_, *multiparams, **params): ...
+    @overload
+    def scalar(
+        self,
+        object_: ClauseElement | FunctionElement | DDLElement | DefaultGenerator | Compiled,
+        *multiparams: Mapping[str, Any],
+        **params: Any,
+    ) -> Any: ...
+    @overload
+    def scalar(self, object_: str, *multiparams: Any | tuple[Any, ...] | Mapping[str, Any], **params: Any) -> Any: ...
     def scalars(self, object_, *multiparams, **params): ...
-    def execute(self, statement, *multiparams, **params): ...
-    def exec_driver_sql(self, statement, parameters: Any | None = ..., execution_options: Any | None = ...): ...
-    def transaction(self, callable_, *args, **kwargs): ...
-    def run_callable(self, callable_, *args, **kwargs): ...
+    @overload  # type: ignore[override]
```

Comment:
In what way is this incompatible with `Connectable`?

Reply:
Different argument names.